pkgs/applications/virtualization/singularity/generic.nix

```diff
@@ -0,0 +1,199 @@
+      fi
+    done
+
+    # Try to make use of the existing configureFlags and configureFlagsArray
+    # Code borrowed from pkgs/stdenv/generic/setup.sh configurePhase()
+
+    # set to empty if unset
+    : ''${configureScript=}
+    : ''${configureFlags=}
+
+    if [[ -z "''${dontAddPrefix:-}" && -n "$prefix" ]]; then
+        configureFlags="''${prefixKey:---prefix=}$prefix $configureFlags"
+    fi
+
+    # Old bash empty array hack
+    # shellcheck disable=SC2086
+    flagsArray=( $configureFlags "''${configureFlagsArray[@]}" )
+    echoCmd 'configure flags' "''${flagsArray[@]}"
+    # shellcheck disable=SC2086
+    $configureScript -V ${version} "''${flagsArray[@]}"
+    unset flagsArray
+
+    # End of the code from pkgs/stdenv/generic/setup.sh configurPhase
+
+    # Don't install SUID binaries
+    sed -i 's/-m 4755/-m 755/g' builddir/Makefile
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make -C builddir
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    make -C builddir install LOCALSTATEDIR=$out/var
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    # Explicitly configure paths in the config file
+    sed -i 's|^# mksquashfs path =.*$|mksquashfs path = ${lib.makeBinPath [squashfsTools]}/mksquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# unsquashfs path =.*$|unsquashfs path = ${lib.makeBinPath [squashfsTools]}/unsquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# cryptsetup path =.*$|cryptsetup path = ${lib.makeBinPath [cryptsetup]}/cryptsetup|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# go path =.*$|go path = ${lib.makeBinPath [go]}/go|' $out/etc/${projectName}/${projectName}.conf
+    ${lib.optionalString enableNvidiaContainerCli ''
+      sed -i 's|^# nvidia-container-cli path =.*$|nvidia-container-cli path = ${lib.makeBinPath [nvidia-docker]}/nvidia-container-cli|' $out/etc/${projectName}/${projectName}.conf
+    ''}
+
+    ${lib.optionalString (removeCompat && (projectName != "singularity")) ''
+      while IFS= read -r -d $'\0' filePath; do
+        if [ -f "$(realpath "$filePath")" ]; then
+          fileDir="$(dirname "$filePath")"
+          fileName="$(basename "$filePath")"
+          fileNameReplaced="''${fileName/singularity/${projectName}}"
+          if [ -e "$fileDir/$fileNameReplaced" ] && [ "$(realpath "$filePath")" == "$(realpath "$fileDir/$fileNameReplaced")" ]; then
+            unlink "$filePath"
+          fi
+        fi
+      done < <(find "$out" -mindepth 1 -type l -name "*singularity*" -print0)
+      for filePath in "$out"/share/*-completion/completions/singularity; do
+        if [ -e "$filePath" ]; then
+          rm "$filePath"
+        fi
+      done
+    ''}
+
+    ${lib.optionalString (!withoutSuid) ''
+      chmod 755 $out/libexec/${projectName}/bin/starter-suid
+    ''}
```

Comment:
```suggestion
```
Is the executable bit missing? Then we should do ``chmod +x``. All files in the nix store are always world readable, so the X55 part is not doing anything. Also removing suid bits is not necessary.